### PR TITLE
✨  📊 pew: Update Same-sex marriage around the world - Pew Research Center

### PR DESCRIPTION
**Pull request by paarriagadap**
From https://github.com/owid/owid-issues/issues/1229

Main LGBT+ rights issue here: https://github.com/owid/owid-issues/issues/1556

### CODE REVIEW

File: etl/steps/data/garden/pew/2024-06-03/same_sex_marriage.meta.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      legal_status:
+        title: Legal status of same-sex marriage
+        unit: ""
+        short_unit: ""
+        description_short: Describes the legislation status of same-sex marriage. This is marriage and marriage recognition between two people of the same biological sex or gender identity.
+        presentation:
+          title_public: Legal status of same-sex marriage
+        display:
+          name: Legal status of same-sex marriage
+          numDecimalPlaces: 0
```
Comment (spoonerf) [inline etl/steps/data/garden/pew/2024-06-03/same_sex_marriage.meta.yml:38]: Not a problem, but you could set `display.numDecimalPlaces` and `display.tolerance` in the `common` section at the top.

File: pyproject.toml [hunk 2]
```diff
@@ -77,7 +77,7 @@ slack-sdk = "^3.26.2"
 joblib = "^1.3.2"
 
 [tool.poetry.group.wizard.dependencies]
-streamlit-nightly = "^1.35.1.dev20240610"
+streamlit-nightly = "1.35.1.dev20240610"
```
Comment (spoonerf) [inline pyproject.toml:80]: Intentional?

File: poetry.lock [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# This file is automatically @generated by Poetry 1.8.2 and should not be changed by hand.
```
Comment (spoonerf) [inline poetry.lock:1]: Do you definitely want to merge this?

File: etl/steps/data/garden/pew/2024-06-03/same_sex_marriage.py [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+    # Fill data forward after the first incidence of "Legal" in each country
+    tb["legal_status"] = tb.groupby("country")["legal_status"].ffill()
+
+    # Add "Not legal" for missing legal status
+    tb.loc[tb["legal_status"].isnull(), "legal_status"] = "Not legal"
+
+    return tb
+
+
+def add_country_counts_and_population_by_status(
```
Comment (spoonerf) [inline etl/steps/data/garden/pew/2024-06-03/same_sex_marriage.py:108]: It's a little hard to follow what's going on in here. Perhaps it might be better as a few smaller functions. Also would be good to add some sense checks/asserts at a few critical points.

**Review by spoonerf — APPROVED** [review]
Looks good, just a few fairly minor points.

File: pyproject.toml (hunk 2 above)
Comment (paarriagadap) [reply pyproject.toml:80]: .@lucasrodes added that one because of problems in Streamlit versions. It is by owidbot so it should be fine (Lucas, let us know otherwise)

File: etl/steps/data/garden/pew/2024-06-03/same_sex_marriage.py (hunk 4 above)
Comment (paarriagadap) [reply etl/steps/data/garden/pew/2024-06-03/same_sex_marriage.py:108]: Thanks Fiona. I am thinking I can adapt this function and publish it on `geo`, because it has been very useful to construct aggregates from the categories of categorical variables. So I will hold changes here to adapt it more generally and publish it for the rest.

**Comment by paarriagadap** [comment]
Thank you, @spoonerf !
